### PR TITLE
Fix text coverage analysis for jvm projects

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR fixes line calculation for JVM projects in textcov analysis. It also filter out fuzzer classes from the coverage analysis to decrease noise.

### CODE REVIEW

File: experiment/textcov.py [hunk 1]
```diff
@@ -315,6 +319,14 @@ def covered_lines(self):
   def total_lines(self):
     return sum(len(f.lines) for f in self.functions.values())
 
+  def is_fuzzer_class(self, class_item) -> bool:
+    """Determine if the class_item is a fuzzer class."""
+    method_elements = class_item.find('./method[@name=\"fuzzerTestOneInput\"]')
```
Comment (DavidKorczynski) [inline experiment/textcov.py:324]: Can this function be shortened to `return bool(class_item.find('./method[@name=\"fuzzerTestOneInput\"]'))`?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: experiment/textcov.py (hunk 1 above)
Comment (arthurscchan) [reply experiment/textcov.py]: Yes. Done.

File: experiment/evaluator.py [hunk 2]
```diff
@@ -332,16 +332,21 @@ def check_target(self, ai_binary, target_path: str) -> Result:
...
     if self.benchmark.language == 'jvm':
-      # The summary.json generated from Jacoco.xml report
-      # of JVM project does not have total lines information.
-      # This fix Use the total lines calculation from the
-      # jacoco.xml report of the current run directly for
-      # JVM projects.
-      total_lines = run_result.coverage.total_lines
-    else:
-      total_lines = _compute_total_lines_without_fuzz_targets(
-          coverage_summary, generated_target_name)
```
Comment (DavidKorczynski) [inline experiment/evaluator.py:344]: Is it intentional to remove the `else` part here?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: experiment/evaluator.py (hunk 2 above)
Comment (DavidKorczynski) [reply experiment/evaluator.py:344]: Ah sorry, you added this above.

**Review by DavidKorczynski — APPROVED** [review]

**Comment by DavidKorczynski** [comment]
Experiment done in: https://github.com/google/oss-fuzz-gen/pull/309

Experiment is looking good: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-06-07-309-dk-test-jvm3-jvm/index.html

**Pull request merged**